pkg/controllermanager/controller/project/project_activity_control.go

```diff
@@ -17,89 +17,226 @@ package project
+	// skip it. This is to prevent unnecessary reconciliations in case of GCM restart.
+	if objMeta.GetCreationTimestamp().Add(time.Hour).UTC().Before(c.clock.Now().UTC()) {
+		return
 	}
 
-	timestamp := obj.GetCreationTimestamp()
-	if project.Status.LastActivityTimestamp == nil {
-		shoots := &gardencorev1beta1.ShootList{}
-		if err := r.gardenClient.List(ctx, shoots, client.InNamespace(*project.Spec.Namespace)); err != nil {
-			return reconcile.Result{}, fmt.Errorf("failed to list shoots in project namespace %q: %w", *project.Spec.Namespace, err)
-		}
-		for _, shoot := range shoots.Items {
-			if shoot.GetCreationTimestamp().UTC().After(timestamp.UTC()) {
-				timestamp = shoot.GetCreationTimestamp()
-			}
-		}
+	key := c.getProjectKey(ctx, objMeta.GetNamespace())
+	if key == "" {
+		return
 	}
 
-	log.Info("Updating Project's lastActivityTimestamp", "lastActivityTimestamp", timestamp)
-	if err := updateStatus(ctx, r.gardenClient, project, func() {
-		project.Status.LastActivityTimestamp = &timestamp
-	}); err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed updating Project's lastActivityTimestamp: %w", err)
+	c.projectActivityQueue.Add(key)
+}
+
+func (c *Controller) projectActivityObjectDelete(ctx context.Context, obj interface{}) {
+	objMeta, err := meta.Accessor(obj)
+	if err != nil {
+		return
 	}
-	return reconcile.Result{}, nil
+
+	key := c.getProjectKey(ctx, objMeta.GetNamespace())
+	if key == "" {
+		return
+	}
+
+	c.projectActivityQueue.Add(key)
 }
 
-func (c *Controller) updateShootActivity(obj interface{}) {
-	key, err := cache.MetaNamespaceKeyFunc(obj)
+func (c *Controller) projectActivityObjectWithLabelAdd(ctx context.Context, obj interface{}) {
+	objMeta, err := meta.Accessor(obj)
 	if err != nil {
-		c.log.Error(err, "Couldn't get key for object", "object", obj)
 		return
 	}
-	c.projectShootActivityQueue.Add(key)
+
+	// skip queueing if the object(secret or quota) doesn't have the "referred by a secretbinding" label
+	if _, hasLabel := objMeta.GetLabels()[v1beta1constants.LabelSecretBindingReference]; !hasLabel {
+		return
+	}
+
+	// If the creationTimestamp of the object is less than 1 hour (experimental) from current time,
+	// skip it. This is to prevent unnecessary reconciliations in case of GCM restart.
+	if objMeta.GetCreationTimestamp().Add(time.Hour).UTC().Before(c.clock.Now().UTC()) {
+		return
+	}
+
+	key := c.getProjectKey(ctx, objMeta.GetNamespace())
+	if key == "" {
+		return
+	}
+
+	c.projectActivityQueue.Add(key)
+}
+
+func (c *Controller) projectActivityObjectWithLabelUpdate(ctx context.Context, oldObj, newObj interface{}) {
+	_, err := meta.Accessor(oldObj)
+	if err != nil {
+		return
+	}
+	newObjMeta, err := meta.Accessor(newObj)
+	if err != nil {
+		return
+	}
+
+	// skip queueing if the object(secret or quota) doesn't have the "referred by a secretbinding" label
+	if _, hasLabel := newObjMeta.GetLabels()[v1beta1constants.LabelSecretBindingReference]; !hasLabel {
```

Comment:
We should skip only if neither `oldObj` nor `newObj` have the label.
Only `oldObj` but not `newObj` having the label would be kind of the delete case for secretbindings.

Reply:
> Only oldObj but not newObj having the label would be kind of the delete case for secretbindings. 

What do you mean?